app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityController.scala

```diff
@@ -372,15 +374,22 @@ class BusinessEntityController @Inject() (
     )
   )
 
+  private val isEoriLengthValid = Constraint[String] { eori: String =>
```

Comment:
You could move this into a form provider so it can be easily tested in isolation like `ClaimEoriFormProvider`.

Reply:
I have already added a test case for this check in controller. I feel a separate form provider is not needed just to check this functionality. We can consider it later if more validation comes up in future.